xdsl/transforms/canonicalization_patterns/riscv.py

```diff
@@ -471,3 +471,39 @@ def match_and_rewrite(
                     comment=op.comment,
                 ),
             )
+
+
+class GetZeroRegister(RewritePattern):
+    """
+    The canonical form of an operation that creates a zero register is `li 0`, so that the
+    downstream patterns can assume that any constant is created by `li`, and don't need
+    more sophisticated checks.
+    To revert to getting the zero register, use `riscv-get-zero-register`
```

Comment:
Hmmmmm, this feels like the wrong approach to me. I'd rather factor out the code that does the cnostant inferrence to handle `zero` registers.

Reply:
I'm not sure that any approach is the wrong aproach, TBH. My understanding of canonicalization is that it's supposed to make things more uniform, as opposed to more efficient. This pass satisfies this quality, in my understanding, making the statement "if it's constant, it's LiOp` true. This feels conceptually simpler than a helper that gets a constant value from a possible range of operations. Ultimately, I'm not sure that either approach is wrong, I think either could work. Am I missing something?